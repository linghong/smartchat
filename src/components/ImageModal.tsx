// components/ImageModal.tsx
import React from 'react';
import Image from 'next/image';

interface ImageModalProps {
  src: string;
  onClose: () => void;
}

const ImageModal: React.FC<ImageModalProps> = ({ src, onClose }) => {
  return (
    <div
      className="w-full fixed p-10 inset-0 bg-slate-500 bg-opacity-90 flex items-center justify-center z-50"
      role="dialog"
      aria-modal="true"
      aria-labelledby="modalTitle"
      aria-describedby="Extended view of the uploaded image"
    >
      <Image
        src={src}
        fill
        style={{ objectFit: 'contain' }}
        alt="Expanded view"
        className="p-6 max-w-full max-h-full"
      />
      <button
        className="absolute top-9 right-9 text-white text-3xl"
        onClick={e => {
          e.stopPropagation();
          onClose();
        }}
        aria-label="Close image view"
      >
        &times;
      </button>
    </div>
  );
};

export default ImageModal;
