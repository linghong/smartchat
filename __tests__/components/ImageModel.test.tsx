import React from 'react'
import { render, screen, fireEvent, RenderResult } from '@testing-library/react'
import '@testing-library/jest-dom'

import ImageModal from '@/src/components/ImageModal'

describe('ImageModal', () => {
  const src = 'http://example.com/image.jpg'
  const onClose = jest.fn()
  let component: RenderResult

  beforeEach(() => {
    component = render(<ImageModal src={src} onClose={onClose} />)
  })

  it('should render the modal with the image', () => {
    const image = screen.getByAltText('Expanded view')

    expect(image).toBeInTheDocument()
    // Note: The actual `src` attribute of the `Image` component is dynamically generated by Next.js
    // We check if the src includes the expected URL part
    expect(image).toHaveAttribute(
      'src',
      expect.stringContaining(encodeURIComponent(src))
    )
  })

  it('should call onClose when the close button is clicked', () => {
    const closeButton = screen.getByRole('button', {
      name: /close image view/i
    })
    fireEvent.click(closeButton)

    expect(onClose).toHaveBeenCalledTimes(1)
  })

  it('should have the correct accessibility attributes', () => {
    const dialog = screen.getByRole('dialog')

    expect(dialog).toHaveAttribute('aria-modal', 'true')
    expect(dialog).toHaveAttribute('aria-labelledby', 'modalTitle')
    expect(dialog).toHaveAttribute(
      'aria-describedby',
      'Extended view of the uploaded image'
    )
  })

  it('should match snapshot', () => {
    const { asFragment } = component
    expect(asFragment()).toMatchSnapshot()
  })
})
