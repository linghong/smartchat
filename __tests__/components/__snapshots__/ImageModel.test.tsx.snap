// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageModal should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-describedby="Extended view of the uploaded image"
    aria-labelledby="modalTitle"
    aria-modal="true"
    class="w-full fixed p-10 inset-0 bg-slate-500 bg-opacity-90 flex items-center justify-center z-50"
    role="dialog"
  >
    <img
      alt="Expanded view"
      class="p-6 max-w-full max-h-full"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="100vw"
      src="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=3840&q=75"
      srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 640w, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=750&q=75 750w, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=828&q=75 828w, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2Fexample.com%2Fimage.jpg&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent;"
    />
    <button
      aria-label="Close image view"
      class="absolute top-9 right-9 text-white text-3xl"
    >
      Ã—
    </button>
  </div>
</DocumentFragment>
`;
